choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerPlanDBService.java

```diff
@@ -0,0 +1,616 @@
+        }
+    }
+
+    //-------------------------------------------------------------------------------------------------
+	public ChoreographerPlanListResponseDTO getPlanEntriesResponse(final int page, final int size, final Direction direction, final String sortField) {
+        logger.debug("getPlanEntriesResponse started...");
+
+        final Page<ChoreographerPlan> planEntries = getPlanEntries(page, size, direction, sortField);
+
+        final List<ChoreographerPlanResponseDTO> planDTOs = new ArrayList<>(planEntries.getSize());
+        try {
+        	for (final ChoreographerPlan plan : planEntries) {
+        		planDTOs.add(DTOConverter.convertPlanToPlanResponseDTO(plan, getPlanDetails(plan)));
+        	}
+        } catch (final Exception ex) {
+            logger.debug(ex.getMessage(), ex);
+            throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+        }
+
+        return new ChoreographerPlanListResponseDTO(planDTOs, planEntries.getTotalElements());
+    }
+	
+    //-------------------------------------------------------------------------------------------------
+	public ChoreographerPlan getPlanById(final long id) {
+        logger.debug("getPlanById started...");
+
+        try {
+            final Optional<ChoreographerPlan> planOpt = choreographerPlanRepository.findById(id);
+            if (planOpt.isPresent()) {
+                return planOpt.get();
+            } else {
+                throw new InvalidParameterException("Choreographer Plan with id of '" + id + "' doesn't exist!");
+            }
+        } catch (final InvalidParameterException ex) {
+            throw ex;
+        } catch (final Exception ex) {
+            logger.debug(ex.getMessage(), ex);
+            throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+        }
+    }
+	
+    //-------------------------------------------------------------------------------------------------
+	public ChoreographerPlanResponseDTO getPlanByIdResponse(final long id) {
+        logger.debug("getPlanByIdResponse started...");
+
+        final ChoreographerPlan plan = getPlanById(id);
+        try {
+        	return DTOConverter.convertPlanToPlanResponseDTO(plan, getPlanDetails(plan));
+        } catch (final Exception ex) {
+            logger.debug(ex.getMessage(), ex);
+            throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+        }
+    }
+	
+	//-------------------------------------------------------------------------------------------------
+	@Transactional(rollbackFor = ArrowheadException.class)
+    public void removePlanEntryById(final long id) {
+        logger.debug("removePlanEntryById started...");
+
+        try {
+        	final Optional<ChoreographerPlan> planOpt = choreographerPlanRepository.findById(id);
+            if (planOpt.isEmpty()) {
+                throw new InvalidParameterException("Plan with id of '" + id + "' doesn't exist!");
```

Comment:
why not just return?

Reply:
That's how remove works for example in ServiceRegistryDBService.

Reply:
than maybe I should follow this regarding to the executros too